### ISSUE TITLE
current min Python is 3.8 --> update to 3.11

### ISSUE THREAD

**Issue by jGaboardi** [martinfleis/clustergram#66]
The current minimal supported Python is 3.8[^1][^2]. This [still passing](https://github.com/martinfleis/clustergram/actions/runs/11964347578/job/33356502097), but should we update to Python 3.11 to follow with [SPEC0](https://scientific-python.org/specs/spec-0000/#support-window)?

I'd like to put in a PR if so.

[^1]: [`pyproject.toml`](https://github.com/martinfleis/clustergram/blob/c99e3cd9b230caa467514080f68d600d390f8b28/pyproject.toml#L23)
[^2]: [`ci/envs/minimal.yaml`](https://github.com/martinfleis/clustergram/blob/c99e3cd9b230caa467514080f68d600d390f8b28/ci/envs/minimal.yaml#L5)

**Comment by martinfleis** [comment]
I don't think it hurts to support more if there's no additional cost of doing so. But sticking to 3.8 might not be needed :). Not sure if a PR will make any difference. I don't have plans or a need for a release now so it would not propagate for a while.
